src/Form/vendor-regex.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * Given some ruleSets, create an efficient
+ * lookup system for accessing cached regexes by name.
+ *
+ * @param {VendorRegexConfiguration["regexes"]} ruleSets
+ * @return {{RULES: Record<keyof VendorRegexRules, RegExp | undefined>}}
+ */
+function createCacheableVendorRegexes (ruleSets) {
```

Comment:
This logic comes from the firefox codebase - I've kept it since it efficiently caches the large regexes on demand